### PR TITLE
CHECKOUT-4201 Match SDK types with API payloads

### PR DESCRIPTION
**Pull request by capsula4**
## What?
Update SDK types

## Why?
To reflect better the real object structures.

## Testing / Proof
N/A

@bigcommerce/checkout

### CODE REVIEW

File: src/cart/carts.mock.ts [hunk 1]
```diff
@@ -7,9 +7,10 @@ import { getDigitalItem, getGiftCertificateItem, getPhysicalItem } from './line-
 
 export function getCart(): Cart {
     return {
-        id: 'b20deef40f9699e48671bbc3fef6ca44dc80e3c7',
+    id: 'b20deef40f9699e48671bbc3fef6ca44dc80e3c7',
```
Comment (icatalina) [inline src/cart/carts.mock.ts]: indent

File: src/address/address.ts [hunk 2]
```diff
@@ -16,6 +18,6 @@ export interface AddressRequestBody {
     phone: string;
     customFields: Array<{
         fieldId: string;
-        fieldValue: string;
+        fieldValue: string | number;
```
Comment (icatalina) [inline src/address/address.ts]: This can be an array of values, can't it?

File: src/address/internal-address.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export default interface InternalAddress {
     phone: string;
     customFields: Array<{
         fieldId: string;
-        fieldValue: string;
+        fieldValue: string | number;
```
Comment (icatalina) [inline src/address/internal-address.ts]: Same than above.

File: README.md [hunk 4]
```diff
@@ -285,7 +287,7 @@ If the submission is successful, you should redirect the customer to the order c
 
 #### Finalize order
 
-Also, for some payment methods, the customer may be asked to enter their payment details on an external website. For these methods, you must finalize the order when the customer is redirected back to the checkout page in order to complete the checkout flow. This should be done in the background before you present any checkout information to the customer. 
+Also, for some payment methods, the customer may be asked to enter their payment details on an external website. For these methods, you must finalize the order when the customer is redirected back to the checkout page in order to complete the checkout flow. This should be done in the background before you present any checkout information to the customer.
```
Comment (icatalina) [inline README.md]: Can we break this line, the same way the one above?

File: src/form/form-field.ts [hunk 5]
```diff
@@ -20,3 +90,5 @@ export interface FormFieldItem {
     value: string;
     label: string;
 }
+
+export default FormField;
```
Comment (icatalina) [inline src/form/form-field.ts]: Might be better to export directly in the `FormField` atop.

```ts
export type FormField = SystemFormField | CustomFormField;
```

Comment (capsula4) [reply src/form/form-field.ts]: I still want to keep FormField as the default export, can't do it in one line.

Comment (icatalina) [reply src/form/form-field.ts]: oh, my bad, I missed the default...

**Pull request merged**